framework/src/csifw/CSIService.c

```diff
@@ -0,0 +1,239 @@
+		if (gCSIDataDumpEvent) {
+			CSIFW_RES ret = csi_logs_dumper_send_to_queue(raw_csi_buff, raw_csi_buff_len);
+			if (ret != CSIFW_OK) {
+				CSIFW_LOGE("Failed to send CSI data to CSI_DUMP_DATA_QUEUE");
+			}
+			CSIFW_LOGD("CSI data sent to CSI_DUMP_DATA_QUEUE");
+		}
+		#endif
+	}
+}
+
+CSIFW_RES csi_manager_init(void)
+{
+	CSIFW_LOGI("[CSIFW] initialization function is called\n");
+	return CSIFW_OK;
+}
+
+CSIFW_RES csi_service_init(upd_raw_data_listener raw_callback, upd_parsed_data_listener parsed_callback, unsigned int interval, void* ptr)
+{	
+	#ifdef CONFIG_CSI_DATA_DUMP_OVER_NETWORK
+	int result = mkfifo(CSI_DUMP_DATA_QUEUE_NAME, 0666);
+	if (result < 0 && result != -EEXIST) {
+		CSIFW_LOGE("create CSI_DUMP_DATA_QUEUE fail %d", errno);
+		return CSIFW_ERROR;
+	}
+	gDataDumpDescriptor = open(CSI_DUMP_DATA_QUEUE_NAME, O_WRONLY | O_NONBLOCK);
+	if (gDataDumpDescriptor < 0) {
+		CSIFW_LOGE("open CSI_DUMP_DATA_QUEUE fail %d", errno);
+		unlink(CSI_DUMP_DATA_QUEUE_NAME);
+		return CSIFW_ERROR;
+	}
+	result = csi_logs_dumper_init();
+	if (result != CSIFW_OK) {
+		CSIFW_LOGE("open csi_logs_dumper_init fail");
+		close(gDataDumpDescriptor);
+		unlink(CSI_DUMP_DATA_QUEUE_NAME);
+		return CSIFW_ERROR;
+	}
+	set_event_listener(csi_data_dump_listener);
+	#endif
+
+	g_ptr = ptr;
+	if (g_service_state != CSI_STATE_UNITIALIZED) {
+		CSIFW_LOGE("Already Initialized");
+		return CSIFW_ERROR;
+	}
+
+	CSIFW_RES res = CSIFW_OK;
+	g_raw_callback = raw_callback;
+	g_parsed_callback = parsed_callback;
+	csi_set_data_collect_interval(interval);
+	gParsedDataBufferLen = CSIFW_MAX_RAW_BUFF_LEN;
+	if (g_parsed_callback) {
+		g_parsed_buffptr = (float *)malloc(sizeof(float) * gParsedDataBufferLen);
+		if (!g_parsed_buffptr) {
+			CSIFW_LOGE("Failed to allocate memory for parsed data buffer, size: %zu", sizeof(float) * gParsedDataBufferLen);
+			return CSIFW_ERROR;
+		}
+	}
+	g_service_state = CSI_STATE_INITIALIZED;
+
+	CSIFW_RES ret;
+	ret = csi_packet_receiver_init(CSIRawDataListener);
+
+	if (ret != CSIFW_OK) {
+		CSIFW_LOGE("csi_packet_receiver_init failed");
+		return CSIFW_ERROR;
+	}
+	CSIFW_LOGD("CSI Packet receiver init done");
+
+	return res;
+}
+
+CSIFW_RES csi_set_data_collect_interval(unsigned interval)
+{
+	return ping_generator_change_interval(interval);
+}
+
+CSIFW_RES csi_service_start()
+{	
+	if (g_service_state == CSI_COLLECT_STATE_STARTED) {
+		CSIFW_LOGE(" Already running");
+		return CSIFW_ERROR;
+	}
+	csi_packet_receiver_start_collect(&g_csifw_config);
+	ping_generator_start();
+	g_service_state = CSI_COLLECT_STATE_STARTED;
+	return CSIFW_OK;
+}
+
+CSIFW_RES csi_service_stop(CSIFW_REASON reason)
+{
+	if (g_service_state != CSI_COLLECT_STATE_STARTED) {
+		CSIFW_LOGE("Already stopped");
+		g_service_state = CSI_COLLECT_STATE_STOPPED;
+		return CSIFW_ERROR;
+	}
+	g_service_state = CSI_COLLECT_STATE_STOPPED;
+	if (reason == CSIFW_WIFI_DISCONNECTED)
+	{
+		CSIFW_LOGI("[CSIFW_REASON] CSIFW_WIFI_DISCONNECTED %d", reason);
```

Comment:
1. Is there any reason why you print this message only in wifi disconnected?
2. The message `CSIFW_WIFI_DISCONNECTED` and `%d` are duplicated. Is it necessary?